tests/test_project_time.py

```diff
@@ -0,0 +1,251 @@
+)
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.fixture
+def registry_mgr():
+    return RegistryManager.load(REGISTRY_PATH, offline_mode=True)
+
+
+def test_no_unexpected_timezone():
+    for tzo in TimeZone:
+        if tzo in [TimeZone.NONE, TimeZone.LOCAL]:
+            assert tzo.is_standard() + tzo.is_prevailing() == 0
+        else:
+            assert (
+                tzo.is_standard() + tzo.is_prevailing() == 1
+            ), f"{tzo} can either be prevailing or standard"
+
+
+def test_convert_to_project_time(registry_mgr):
+    project_id = "dsgrid_conus_2022"
+    project = registry_mgr.project_manager.load_project(project_id)
+
+    dataset_id = "conus_2022_reference_resstock"
+    project.load_dataset(dataset_id)
+    resstock = project.get_dataset(dataset_id)
+
+    dataset_id = "conus_2022_reference_comstock"
+    project.load_dataset(dataset_id)
+    comstock = project.get_dataset(dataset_id)
+
+    dataset_id = "tempo_conus_2022"
+    project.load_dataset(dataset_id)
+    tempo = project.get_dataset(dataset_id)
+
+    # different ways to access project_time_dim:
+    project_time_dim = project.config.get_base_dimension(
+        DimensionType.TIME
+    )  # or tempo._handler._project_time_dim
+    resstock_time_dim = resstock._handler.config.get_dimension(DimensionType.TIME)
+    comstock_time_dim = comstock._handler.config.get_dimension(DimensionType.TIME)
+    tempo_time_dim = tempo._handler.config.get_dimension(DimensionType.TIME)
+
+    # [1] test get_time_dataframe()
+    check_time_dataframe(project_time_dim)
+    check_time_dataframe(resstock_time_dim)
+    check_time_dataframe(comstock_time_dim)
+    tempo_time_dim.get_time_dataframe()
+    # TODO: could add test for annual_time_dimension_config when AEO data is ready
+
+    # [2] test convert_dataframe()
+    # tempo time explosion
+    # Test 1: feed in load_data and load_data_lookup separately
+    tempo_time_dim.convert_dataframe(
+        df=tempo.load_data,
+        project_time_dim=project_time_dim,
+        df_meta=tempo._handler._add_time_zone(tempo.load_data_lookup),
+    )
+    # Test 2: feed in load_data and lookup combined as tempo_data
+    tempo_data = tempo._handler._add_time_zone(tempo.load_data_lookup)
+    tempo_data = tempo.load_data.join(tempo_data, on="id")
+    tempo_data_mapped = tempo_time_dim.convert_dataframe(
+        df=tempo_data,
+        project_time_dim=project_time_dim,
+    )
+    check_exploded_tempo_time(project_time_dim, tempo_data_mapped)
+    check_tempo_load_sum(project_time_dim, tempo, tempo_data, tempo_data_mapped)
+
+    # comstock time conversion
+    comstock_data = comstock._handler._add_time_zone(comstock.load_data_lookup)
+    comstock_data = comstock.load_data.join(comstock_data, on="id")
+    comstock_data = comstock_time_dim.convert_dataframe(
+        df=comstock_data,
+        project_time_dim=project_time_dim,
+    )
+    # TODO: add test for annual_time_dimension_config, res/com with local timezones
```

Comment:
Would these be in this same test or a different test? I don't really get what needs to be checked here. If we can't write these tests now, maybe document in a git issue?